Sources/APIProvider.swift

```diff
@@ -163,6 +163,10 @@ private extension APIProvider {
                 return .failure(Error.requestFailure(response.statusCode, response.data))
             }
 
+            if T.self is Data.Type {
+                return .success(data as! T)
```

Comment:
This is a bit nicer and removes the force unwrap!

```suggestion
            if let data as? T {
                return .success(data)
```

Reply:
@AvdLee Thanks for the reply, I have tried this but it gave me a syntax error and couldn't compile, did you try this before?

Reply:
`let data = data as? T` could work how @AvdLee expected the suggestion would work

Reply:
Thanks mate, it works

Reply:
Oops, my bad. Thanks for correcting me @kinnarr!